src/storage/src/table/mod.rs

```diff
@@ -125,6 +124,33 @@ where
     }
 }
 
+/// Collects data chunks from stream of rows.
+pub async fn collect_data_chunk_with_builder<E, S>(
+    stream: &mut S,
+    chunk_size: Option<usize>,
+    builder: &mut DataChunkBuilder,
+) -> Result<Option<DataChunk>, E>
+where
+    S: Stream<Item = Result<OwnedRow, E>> + Unpin,
+{
+    for _ in 0..chunk_size.unwrap_or(usize::MAX) {
```

Comment:
I believe the interface is somehow confusing. There's also a "chunk size" implied in the capacity of the passed `builder`. If it does not match the `chunk_size` argument, there will possibly be panic.

From my perspective, we don't need this `chunk_size` anymore. If the caller wants an unlimited chunk size, he can pass a `builder` with `usize::MAX` as the capacity (or make it an `Option`).

Reply:
Agree, I guess the builder should be initialized with chunk size instead.